pkg/controller/elasticsearch/remotecluster/remoteca/controller.go

```diff
@@ -0,0 +1,474 @@
+		return results.WithError(err)
+	}
+
+	return nil
+}
+
+// reconcileRemoteCA copies certificates authorities across 2 clusters
+func reconcileRemoteCA(
+	ctx context.Context,
+	c k8s.Client,
+	owner *esv1.Elasticsearch,
+	remote types.NamespacedName,
+	remoteCA []byte,
+) error {
+	span, _ := apm.StartSpan(ctx, "reconcile_remote_ca", tracing.SpanTypeApp)
+	defer span.End()
+
+	// Define the desired remote CA object, it lives in the remote namespace.
+	expected := corev1.Secret{
+		ObjectMeta: remoteCAObjectMeta(remoteCASecretName(owner.Name, remote), owner, remote),
+		Data: map[string][]byte{
+			certificates.CAFileName: remoteCA,
+		},
+	}
+
+	var reconciled corev1.Secret
+	return reconciler.ReconcileResource(reconciler.Params{
+		Client:     c,
+		Scheme:     scheme.Scheme,
+		Owner:      owner,
+		Expected:   &expected,
+		Reconciled: &reconciled,
+		NeedsUpdate: func() bool {
+			return !maps.IsSubset(expected.Labels, reconciled.Labels) || !reflect.DeepEqual(expected.Data, reconciled.Data)
+		},
+		UpdateReconciled: func() {
+			reconciled.Labels = maps.Merge(reconciled.Labels, expected.Labels)
+			reconciled.Data = expected.Data
+		},
+	})
+}
+
+func caCertMissingError(location string, cluster types.NamespacedName) string {
+	return fmt.Sprintf(
+		CaCertMissingError,
+		location,
+		cluster.Namespace,
+		cluster.Name,
+	)
+}
+
+// getExpectedRemoteCertificateAuthorities returns all the remote cluster keys for which a remote ca should created
+func getExpectedRemoteCertificateAuthorities(
+	ctx context.Context,
+	c k8s.Client,
+	associatedEs *esv1.Elasticsearch,
+) (map[types.NamespacedName]struct{}, error) {
+	span, _ := apm.StartSpan(ctx, "get_expected_remote_ca", tracing.SpanTypeApp)
+	defer span.End()
+	expectedRemoteClusters := make(map[types.NamespacedName]struct{})
+
+	// Add remote clusters declared in the Spec
+	for _, remoteCluster := range associatedEs.Spec.RemoteClusters.K8sLocal {
+		if !remoteCluster.IsDefined() {
+			continue
+		}
+		esRef := remoteCluster.ElasticsearchRef.WithDefaultNamespace(associatedEs.Namespace)
+		expectedRemoteClusters[esRef.NamespacedName()] = struct{}{}
+	}
+
+	var list esv1.ElasticsearchList
+	if err := c.List(&list, &client.ListOptions{}); err != nil {
+		return nil, err
+	}
+
+	// Seek for Elasticsearch resources where this cluster is declared as a remote cluster
+	for _, es := range list.Items {
+		for _, remoteCluster := range es.Spec.RemoteClusters.K8sLocal {
+			if !remoteCluster.IsDefined() {
+				continue
+			}
+			esRef := remoteCluster.ElasticsearchRef.WithDefaultNamespace(es.Namespace)
+			if esRef.Namespace == associatedEs.Namespace &&
+				esRef.Name == associatedEs.Name {
+				expectedRemoteClusters[k8s.ExtractNamespacedName(&es)] = struct{}{}
+			}
+		}
+	}
+
+	return expectedRemoteClusters, nil
+}
+
+// getCurrentRemoteCertificateAuthorities returns all the remote cluster keys for which a remote ca exists
+func getCurrentRemoteCertificateAuthorities(
```

Comment:
func name suggestion: `func clustersWithRemoteCAs()`

Reply:
`getCurrentRemoteCertificateAuthorities` is the counter part of `getExpectedRemoteCertificateAuthorities`. I think it is more explicit if we keep the current naming (I will rename it to getActualRemoteCertificateAuthorities).